### PR TITLE
proposal: add device allocation apis

### PR DESCRIPTION
**Pull request by eahydra**
### Ⅰ. Describe what this PR does

This proposal introduces new APIs to define anticipated device allocation strategies. One API set supports joint allocation scenarios for devices like GPUs and NICs according to the hardware topology. The other set offers device allocation hints, detailing specific allocation rules to guide scheduler decisions.

### CODE REVIEW

**Comment by eahydra** [comment]
I have updated more details about the new APIs. PTAL @hormes @FillZpp @jasonliu747 @ZiMengSheng

File: docs/proposals/scheduling/20230803-device-allocate-hint-apis.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+# Device Allocate Hint API
+
+## Summary
+
+This proposal introduces new APIs to define anticipated device allocation strategies. One API set supports joint allocation scenarios for devices like GPUs and NICs according to the hardware topology. The other set offers device allocation hints, detailing specific allocation rules to guide scheduler decisions.
+
+## Motivation
+
+Current mechanisms for RDMA device allocation are independent and may require differentiation between Physical Functions (PFs) and Virtual Functions (VFs) when SR-IOV is enabled. Users often need to allocate RDMA devices based on their proximity to GPUs to optimize communication in AI model training scenarios. This requirement presents challenges in Kubernetes environments that do not support Device Resource Allocation (DRA).
```
Comment (ZiMengSheng) [inline docs/proposals/scheduling/20230803-device-allocate-hint-apis.md:46]: 这里是不是想说 Dynamic Resource Allocation

Comment (ZiMengSheng) [reply docs/proposals/scheduling/20230803-device-allocate-hint-apis.md:46]: not device  but dynamic

**Review by jasonliu747 — APPROVED** [review]

**Pull request merged**